src/main/java/encoder/Encoder.java

```diff
@@ -0,0 +1,80 @@
+package encoder;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import proto.SensorData.Pose;
+
+/** Change file format from .txt to proto format. */
+public class Encoder {
+  /*
+   * Convert a given .txt that contains a header line specifying file contents
+   * into the defined protobuf format.
+   */
+  public static void encode(InputStream inStream, OutputStream outStream) {
+    // Set an upper threshold by default on the maximum number of fields, this
+    // will make the call to split faster.
+    int maxFields = 20;
+    // Split the input based on spaces, any number of spaces is allowed.
+
+    try (BufferedReader reader = new BufferedReader(new InputStreamReader(inStream, "UTF-8"))) {
+      // The first thing we want to do is read the header line, this will tell
+      // us the names of the fields, and how many input fields that there are.
+      String headLine = reader.readLine();
+      // Check to ensure that a header exists in our input file.
+      if (headLine == null) {
+        System.err.println("Invalid file format: no header provided");
+        closeStream(outStream);
+        return;
+      }
+      String[] header = parseFields(headLine, maxFields);
```

Comment:
`header` is never used except to set `maxFields`—but you already know how many fields to expect, since you're only accepting one input format right now. I think you should ignore the header line and document on this function the expected sequence of fields in the input rows.